seq2science/util.py

```diff
@@ -407,3 +408,132 @@ def shorten(string, max_length, methods="right"):
         string = string[:ceil(max_length/2)] + string[len(string)-floor(max_length/2):]
 
     return string
+
+
+class CaptureStdout(list):
+    """
+    Context manager that somehow manages to capture prints,
+    and not snakemake log
+    """
+    def __enter__(self):
+        self._stdout = sys.stdout
+        sys.stdout = self._stringio = StringIO()
+        return self
+
+    def __exit__(self, *args):
+        self.extend(self._stringio.getvalue().splitlines())
+        del self._stringio  # free up some memory
+        sys.stdout = self._stdout
+
+
+class CaptureStderr(list):
+    """
+    Context manager that somehow manages to capture prints,
+    and not snakemake log
+    """
+    def __enter__(self):
+        self._stderr = sys.stderr
+        sys.stderr = self._stringio = StringIO()
+        return self
+
+    def __exit__(self, *args):
+        self.extend(self._stringio.getvalue().splitlines())
+        del self._stringio  # free up some memory
+        sys.stderr = self._stderr
+
+
+def parse_samples(config):
```

Comment:
this whole section is still present in config_generic.smk, and the function is unused(?)

Reply:
!